contracts/SSVNetwork.sol

```diff
@@ -239,413 +239,271 @@ contract SSVNetwork is OwnableUpgradeable, ISSVNetwork {
+    function reactivatePod(
+        uint64[] memory operatorIds,
+        uint256 amount,
+        Pod memory pod
+    ) external override {
 
         if (!pod.disabled) {
             revert PodAlreadyEnabled();
         }
 
-        _deposit(msg.sender, clusterId, amount.shrink()); // 28k gas usage
-
-        uint64[] memory operatorIds = _clusters[clusterId].operatorIds;
-
-        { // 112k gas usage
-            for (uint64 index = 0; index < operatorIds.length; ++index) {
-                uint64 id = operatorIds[index];
-                Operator memory operator = _operators[id];
-
+        uint64 podIndex;
+        uint64 burnRate;
+        {
+            for (uint8 i = 0; i < operatorIds.length; ++i) {
+                Operator memory operator = _operators[operatorIds[i]];
                 if (operator.owner != address(0)) {
                     operator.snapshot = _getSnapshot(operator, uint64(block.number));
                     operator.validatorCount += pod.validatorCount;
-                    _operators[operatorIds[index]] = operator;
+                    podIndex += operator.snapshot.index + (uint64(block.number) - operator.snapshot.block) * operator.fee;
+                    burnRate += operator.fee;
+                    _operators[operatorIds[i]] = operator;
                 }
             }
         }
 
-        { // 30k gas usage
+        bytes32 hashedPod = _validateHashedPod(msg.sender, operatorIds, pod);
+
+        if (amount > 0) {
+            _deposit(msg.sender, hashedPod, amount.shrink());
+            pod.balance += amount.shrink();
+        }
+
+        pod.disabled = false;
+        pod.index = podIndex;
+
+        pod = _updatePodData(pod, podIndex, 0);
+
+        {
             DAO memory dao = _dao;
             dao = _updateDAOEarnings(dao);
             dao.validatorCount += pod.validatorCount;
             _dao = dao;
         }
 
-        pod.disabled = false;
-        pod = _updatePodData(clusterId, 0, hashedPod, 0); // 16k gas usage
-        _pods[hashedPod] = pod;
+        if (_liquidatable(_podBalance(pod, podIndex), pod.validatorCount, burnRate)) {
+            revert NotEnoughBalance();
+        }
 
-        emit PodEnabled(msg.sender, clusterId);
+        _pods[hashedPod] = keccak256(abi.encodePacked(pod.validatorCount, pod.networkFee, pod.networkFeeIndex, pod.index, pod.balance, pod.disabled ));
+
+        emit PodEnabled(msg.sender, operatorIds);
+        emit PodMetadataUpdated(msg.sender, operatorIds, pod);
     }
-    */
 
     /******************************/
     /* Balance External Functions */
     /******************************/
 
-    /*
-    function deposit(address owner, bytes32 clusterId, uint256 amount) external override {
-        _validateClusterId(clusterId);
+    function deposit(
+        address owner,
+        uint64[] memory operatorIds,
+        uint256 amount,
+        Pod memory pod
+    ) external override {
+        _validatePodIsNotLiquidated(pod);
```

Comment:
Not sure this is correct.
I think you should be allowed to deposit to a liquidated pod.

Reply:
no, if you wanna deposit there is a function reactivatePod with deposit and important logic for that like podIndex recalculation